### ISSUE TITLE
Liked doesn't work

### ISSUE THREAD

**Issue by BasDV** [mopidy/mopidy-soundcloud#79]
Good morning!

I've just installed the Soundcloud extension for Modipy and couldn't get the 'Liked' option to work after authenticating via https://www.mopidy.com/authenticate/.

Groups and Stream work perfectly for example.
Do others also experience this?

Unfortunately I haven't got a log file, weird thing is that log files aren't generated?
Using Mopidy as a service.

**Comment by itsdf** [comment]
Seem to have the same problem running mopdiy + ncmpcpp. Ran a `sudo systemctl status mopidy.service` and found that some souncloud songs aren't found and after that ncmpcpp timeouts. Could you try and check the status after you've tried to access your likes so that we can try and determine if we have the same problem.

```
~ ➤ sudo systemctl status mopidy.service                                                                                                                                                         
● mopidy.service - Mopidy music server
   Loaded: loaded (/lib/systemd/system/mopidy.service; enabled; vendor preset: enabled)
   Active: active (running) since Mon 2016-06-27 15:55:54 BST; 35min ago
  Process: 24361 ExecStartPre=/bin/chown mopidy:audio /var/cache/mopidy (code=exited, status=0/SUCCESS)
  Process: 24358 ExecStartPre=/bin/mkdir -p /var/cache/mopidy (code=exited, status=0/SUCCESS)
 Main PID: 24364 (mopidy)
   CGroup: /system.slice/mopidy.service
           └─24364 /usr/bin/python /usr/bin/mopidy --config /usr/share/mopidy/conf.d:/etc/mopidy/mopidy.conf

Jun 27 16:27:43 zeus mopidy[24364]: INFO     Failed to lookup soundcloud:song/GuABB - The Speech BUY FREE DOWNLOAD 3.250016308: SoundCloud track not found
Jun 27 16:27:44 zeus mopidy[24364]: INFO     Failed to lookup soundcloud:song/Poezia (Original Mix).183101485: SoundCloud track not found
Jun 27 16:27:44 zeus mopidy[24364]: INFO     Failed to lookup soundcloud:song/Dustycloud - Maia (Original Mix).155512174: SoundCloud track not found
Jun 27 16:27:45 zeus mopidy[24364]: INFO     Failed to lookup soundcloud:song/Dimitri Vegas Like Mike vs Steve Aoki - We Are Legend.163517631: SoundCloud track not found
```

**Comment by BasDV** [comment]
It appears that I've got some sort of Python error:
``Jun 27 19:52:27 modipy-bas mopidy[732]: exec('raise tp, value, tb')
Jun 27 19:52:27 modipy-bas mopidy[732]: File "/usr/lib/python2.7/dist-packages/pykka/actor.py", line 201, in _actor_loop
Jun 27 19:52:27 modipy-bas mopidy[732]: response = self._handle_receive(message)
Jun 27 19:52:27 modipy-bas mopidy[732]: File "/usr/lib/python2.7/dist-packages/pykka/actor.py", line 295, in _handle_receive
Jun 27 19:52:27 modipy-bas mopidy[732]: return callee(*message['args'], **message['kwargs'])
Jun 27 19:52:27 modipy-bas mopidy[732]: File "/usr/lib/python2.7/dist-packages/mopidy_soundcloud/library.py", line 166, in browse
Jun 27 19:52:27 modipy-bas mopidy[732]: return self.list_liked()
Jun 27 19:52:27 modipy-bas mopidy[732]: File "/usr/lib/python2.7/dist-packages/mopidy_soundcloud/library.py", line 78, in list_liked
Jun 27 19:52:27 modipy-bas mopidy[732]: logger.debug('Adding liked track %s to vfs' % data.name)
Jun 27 19:52:27 modipy-bas mopidy[732]: AttributeError: 'list' object has no attribute 'name'


Maybe it got diffrent returned data than it excepted (symbols and such)?

**Comment by DvdChe** [comment]
Hello,

I have exactly the same issue as all of you... 

` 07 17:50:22 raspberrypi mopidy[1510]: exec('raise tp, value, tb')
Jul 07 17:50:22 raspberrypi mopidy[1510]: File "/usr/lib/python2.7/dist-packages/pykka/actor.py", line 201, in _actor_loop
Jul 07 17:50:22 raspberrypi mopidy[1510]: response = self._handle_receive(message)
Jul 07 17:50:22 raspberrypi mopidy[1510]: File "/usr/lib/python2.7/dist-packages/pykka/actor.py", line 295, in _handle_receive
Jul 07 17:50:22 raspberrypi mopidy[1510]: return callee(*message['args'], **message['kwargs'])
Jul 07 17:50:22 raspberrypi mopidy[1510]: File "/usr/lib/python2.7/dist-packages/mopidy_soundcloud/library.py", line 166, in browse
Jul 07 17:50:22 raspberrypi mopidy[1510]: return self.list_liked()
Jul 07 17:50:22 raspberrypi mopidy[1510]: File "/usr/lib/python2.7/dist-packages/mopidy_soundcloud/library.py", line 78, in list_liked
Jul 07 17:50:22 raspberrypi mopidy[1510]: logger.debug('Adding liked track %s to vfs' % data.name)
Jul 07 17:50:22 raspberrypi mopidy[1510]: AttributeError: 'list' object has no attribute 'name'`

**Comment by BasDV** [comment]
I've fixed the problem for the likes not retrieving correctly from Soundcloud. Fix was to update the API URL and searched for the users ID. 

Now I want to add the new file to the Github. Unfortunatly I cannot do that. Does anyone know how this can be done?

**Comment by kingosticks** [comment]
Fork this repository. Make the fix in your forked repository. Create a Pull Request. Unfortunately since mopidy-soundcloud does not currently have a maintainer it may take some time for someone to verify and merge the change.

**Comment by kingosticks** [comment]
Liked and following should have been fixed in #100. Please re-open if this is not he case.
